M/MAiNGO/build_tarballs.jl

```diff
@@ -0,0 +1,66 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "MAiNGO"
+version = v"0.7.2"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://git.rwth-aachen.de/avt-svt/public/maingo.git", "252733413a29dbe5b84a4cdaf53e60e9934f372f"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd maingo/
+git remote set-url origin https://git.rwth-aachen.de/avt-svt/public/maingo.git
+mkdir build
+cd build
+git submodule init
+git submodule update -j 1
+if [[ "${target}" == x86_64-apple-darwin* ]]; then
+    export MACOSX_DEPLOYMENT_TARGET=10.15
```

Comment:
I'm surprised this actually works: by default we use the SDK 10.12, if you really need a different version you need to add that to the sources and replace the default SDK with that one.  If you aren't doing that and the build is successful, maybe 10.15 isn't needed after all?

Reply:
I will add the complete code for MacOS and FreeBSD later once I get access to that machine where this was tried again.
Then we can check if the build succeeds.

From what I read, updating the SDK will likely not be enough, but lets see.

With FreeBSD we will have to force building with GCC. Would that be ok also for the MacOS case?

Reply:
MacOS was added, FreeBSD failed, likely because the way Ipopt is integrated upstream using CMake.